### PR TITLE
Set up prod2 deployment for new TDR production instance (#3517)

### PR DESCRIPTION
**Pull request by danielsotirhos**
https://github.com/DataBiosphere/azul/issues/3517

### Author

- [x] PR title references issue
- [x] PR title matches issue title (preceded by `Fix: ` for bugs)   <sub>or there is a good reason why they're different</sub>
- [x] Title of main commit references issue
- [x] PR is connected to Zenhub issue and description links to issue

### Author (reindex)

- [x] Added `r` tag to commit title                         <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR                           <sub>or this PR does not require reindexing</sub>

### Author (freebies & chains)

- [x] Freebies are blocked on this PR                       <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles              <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in the chain        <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR                <sub>or this PR is not chained to another PR</sub>

### Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst  <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title                         <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR                           <sub>or this PR does not require upgrading</sub>
- [x] Added announcement to PR description                  <sub>or this PR does not require announcement</sub>
- [x] Added checklist items for additional operator tasks   <sub>or this PR does not require additional tasks</sub>

### Author (requirements, before every review)

- [x] Ran `make requirements_update`                        <sub>or this PR does not touch requirements*.txt, common.mk, Makefile and Dockerfile</sub>
- [x] Added `R` tag to commit title                         <sub>or this PR does not touch requirements*.txt</sub>
- [x] Added `reqs` label to PR                              <sub>or this PR does not touch requirements*.txt</sub>

### Author (before every review)

- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>
- [x] Rebased branch on `develop`, squashed old fixups

### Primary reviewer (after approval)

- [x] Commented in issue about demo expectations            <sub>or labelled issue as `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] PR title is appropriate as title of merge commit
- [x] Moved ticket to Approved column
- [x] Assigned PR to an operator

### Operator (before pushing merge the commit)

- [x] Checked `reindex` label and `r` commit title tag
- [x] Checked that demo expectations are clear              <sub>or issue is labeled as `no demo`</sub>
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Pushed PR branch to Github
- [x] Branch pushed to Gitlab and added `sandbox` label     <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passed in sandbox                               <sub>or PR is labeled `no sandbox`</sub>
- [x] Started reindex in `sandbox`                          <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Checked for failures in `sandbox`                     <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Added PR reference to merge commit title
- [x] Collected commit title tags in merge commit title
- [x] Moved linked issue to Merged column
- [ ] Pushed merge commit to Github

### Operator (after pushing the merge commit)

- [ ] Made announcement requested by author                 <sub>or PR description does not contain an announcement</sub>
- [ ] Moved freebies to Merged column                       <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened the PR chain                                <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate       <sub>or or this PR is authored by lead</sub>
- [ ] Pushed merge commit to Gitlab                         <sub>or or merge commit can be pushed later, with another PR</sub>
- [ ] Deleted PR branch from Github and Gitlab
- [ ] Build passes on Gitlab

### Operator (reindex) 

- [ ] Started reindex in target deployment                  <sub>or this PR does not require reindexing</sub>
- [ ] Checked for and triaged indexing failures             <sub>or this PR does not require reindexing</sub>
- [ ] Emptied fail queues in target deployment              <sub>or this PR does not require reindexing</sub>

### Operator

- [ ] Unassigned PR

### CODE REVIEW

File: deployments/prod2/environment.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        # Set variables for the `prod2` (short for production 2) deployment here.
+        #
+        # Only modify this file if you intend to commit those changes. To change the
+        # environment with a setting that's specific to you AND the deployment, create
+        # a environment.local.py right next to this file and make your changes there.
+        # Settings applicable to all environments but specific to you go into
+        # environment.local.py at the project root.
+
+        'AZUL_VERSIONED_BUCKET': 'edu-ucsc-gi-azul-dcp2-prod2-config.{AWS_DEFAULT_REGION}',
+        'AZUL_DOMAIN_NAME': 'azul.data.humancellatlas.org',
```
Comment (hannes-ucsc) [inline deployments/prod2/environment.py:36]: Can't be the same as prod

File: deployments/prod2/environment.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+        'AZUL_DEPLOYMENT_STAGE': 'prod2',
+
+        'AZUL_S3_BUCKET': 'edu-ucsc-gi-azul-dcp2-prod2-storage-{AZUL_DEPLOYMENT_STAGE}',
```
Comment (hannes-ucsc) [inline deployments/prod2/environment.py:40]: The fact that `prod` occurs twice in the bucket name is a bug.

File: deployments/prod2/environment.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+        # This deployment shares an ES domain with `prod`
+        'AZUL_SHARE_ES_DOMAIN': '1',
+        'AZUL_ES_DOMAIN': 'azul-index-prod',
+
+        # $0.372/h × 4 × 24h/d × 30d/mo = $1071.36/mo
+        'AZUL_ES_INSTANCE_TYPE': 'r5.xlarge.elasticsearch',
+        'AZUL_ES_INSTANCE_COUNT': '4',
+
+        'AZUL_ES_VOLUME_SIZE': '128',
```
Comment (hannes-ucsc) [inline deployments/prod2/environment.py:77]: Redundant if sharing is enabled.

File: deployments/prod2/environment.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        'AZUL_DEBUG': '1',
+
+        'AZUL_OWNER': 'hannes@ucsc.edu',
+
+        'AZUL_AWS_ACCOUNT_ID': '542754589326',
+        'AWS_DEFAULT_REGION': 'us-east-1',
+
+        'GOOGLE_PROJECT': 'platform-hca-prod',
+
+        'AZUL_CONTRIBUTION_CONCURRENCY': '300/64',
```
Comment (hannes-ucsc) [inline deployments/prod2/environment.py:88]: Too much for only two snapshots.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
```diff
Index: deployments/prod2/environment.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/deployments/prod2/environment.py b/deployments/prod2/environment.py
--- a/deployments/prod2/environment.py	(revision daab5897fb9c1cc5d10ff439042b87ac8edcff8d)
+++ b/deployments/prod2/environment.py	(date 1636411523093)
@@ -24,7 +24,7 @@
     provide the value.
     """
     return {
-        # Set variables for the `prod2` (short for production 2) deployment here.
+        # Set variables for the `prod2` deployment here.
         #
         # Only modify this file if you intend to commit those changes. To change the
         # environment with a setting that's specific to you AND the deployment, create
@@ -32,12 +32,12 @@
         # Settings applicable to all environments but specific to you go into
         # environment.local.py at the project root.
 
-        'AZUL_VERSIONED_BUCKET': 'edu-ucsc-gi-azul-dcp2-prod2-config.{AWS_DEFAULT_REGION}',
-        'AZUL_DOMAIN_NAME': 'azul.data.humancellatlas.org',
+        'AZUL_VERSIONED_BUCKET': 'edu-ucsc-gi-azul-dcp2-prod-config.{AWS_DEFAULT_REGION}',
+        'AZUL_DOMAIN_NAME': 'azul2.data.humancellatlas.org',
 
         'AZUL_DEPLOYMENT_STAGE': 'prod2',
 
-        'AZUL_S3_BUCKET': 'edu-ucsc-gi-azul-dcp2-prod2-storage-{AZUL_DEPLOYMENT_STAGE}',
+        'AZUL_S3_BUCKET': 'edu-ucsc-gi-azul-dcp2-prod-storage-{AZUL_DEPLOYMENT_STAGE}',
 
         'AZUL_CATALOGS': json.dumps({
             f'{name}': dict(atlas=atlas,
@@ -70,12 +70,6 @@
         'AZUL_SHARE_ES_DOMAIN': '1',
         'AZUL_ES_DOMAIN': 'azul-index-prod',
 
-        # $0.372/h × 4 × 24h/d × 30d/mo = $1071.36/mo
-        'AZUL_ES_INSTANCE_TYPE': 'r5.xlarge.elasticsearch',
-        'AZUL_ES_INSTANCE_COUNT': '4',
-
-        'AZUL_ES_VOLUME_SIZE': '128',
-
         'AZUL_DEBUG': '1',
 
         'AZUL_OWNER': 'hannes@ucsc.edu',
@@ -84,6 +78,4 @@
         'AWS_DEFAULT_REGION': 'us-east-1',
 
         'GOOGLE_PROJECT': 'platform-hca-prod',
-
-        'AZUL_CONTRIBUTION_CONCURRENCY': '300/64',
     }
```

File: deployments/prod2/environment.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            ] for i, name in enumerate(names)
+        }),
+
+        'AZUL_PARTITION_PREFIX_LENGTH': '2',
+
+        'AZUL_TDR_SOURCE_LOCATION': 'US',
+        'AZUL_TDR_SERVICE_URL': 'https://data.terra.bio',
+        'AZUL_SAM_SERVICE_URL': 'https://sam.dsde-prod.broadinstitute.org',
+
+        'AZUL_URL_REDIRECT_BASE_DOMAIN_NAME': 'azul.data.humancellatlas.org',
```
Comment (hannes-ucsc) [inline deployments/prod2/environment.py:66]: As previously mentioned, this domain name can't be the same as for `prod`.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

**Review by hannes-ucsc — APPROVED** [review]
Please coordinate with operator to get this deployed. It's ok to deploy this from the `develop` branch (using the override for `check_branch`). Be careful. Run `make -C terraform plan` first to see what would happen. Make sure I'm around when you do this.

**Comment by hannes-ucsc** [comment]
Operator, this requires reindexing to fill the index.

**Comment by hannes-ucsc** [comment]
@danielsotirhos checklist has holes. You have to either check an item or strike it out.

**Comment by amarjandu** [comment]
https://service.azul2.data.humancellatlas.org/ is live, reindexed without failures

**Pull request merged**